src/manifests_workflow/input_manifests.py

```diff
@@ -84,92 +85,75 @@ def update(
-                            branch="main",
-                        )
-
-                        component_version = component.version
-                        if component_version:
-                            release_version = ".".join(component_version.split(".")[:3])
-                            if release_version not in main_versions.keys():
-                                main_versions[release_version] = []
-                            main_versions[release_version].append(component)
-                            logging.info(f"{component.name}#main is version {release_version} (from {component_version})")
-
-            # summarize
-            logging.info("Found versions on main:")
-            for main_version in main_versions.keys():
-                for component in main_versions[main_version]:
-                    logging.info(f" {component.name}={main_version}")
+                if version not in branch_versions:
+                    branch_versions[version] = branch
 
             # generate new manifests
-            for release_version in sorted(main_versions.keys() - known_versions):
-                self.write_manifest(release_version, main_versions[release_version])
-                self.add_to_cron(release_version)
-                self.add_to_versionincrement_workflow(release_version)
-
-    def create_manifest(self, version: str, components: List = []) -> InputManifest:
-        templates_base_path = os.path.join(self.manifests_path(), "templates")
-        template_version_folder = version.split(".")[0] + ".x"
-        template_full_path = os.path.join(templates_base_path, self.prefix, template_version_folder, "manifest.yml")
-        if not os.path.exists(template_full_path):
-            template_full_path = os.path.join(templates_base_path, self.prefix, "default", "manifest.yml")
+            new_versions = sorted(branch_versions.keys() - known_versions, key=version_parse)
+            logging.info(f"New Versions: {new_versions}")
+            for new_version_entry in new_versions:
+                self.write_manifest(new_version_entry, branch_versions[new_version_entry], known_versions)
+                self.add_to_cron(new_version_entry)
+                self.add_to_versionincrement_workflow(new_version_entry)
+                known_versions.append(new_version_entry)
+
+    def create_manifest(self, version: str, branch: str, known_versions: List[str]) -> InputManifest:
+        # If  : No known_versions manifests exist or new version smaller than the min(known_versions), create new manifests from the templates
+        #       (1.0.0-3.0.0 based on template 1.x-3.x, 4.0.0+ from default.x, previous behavior)
+        # Else: Create new manifests based on the latest version before the new version
+        #       (2.12.1 from 2.12.0, 2.13.0 from 2.12.1, 3.0.0 from 2.13.0, 4.0.0 from 3.0.0, etc.)
+        if not known_versions or Version(version) < Version(min(known_versions)):
```

Comment:
Did not fully grasp what is happening here?
`if not known_versions`: This means the code did not find any existing manifests in the manifest path, right? If yes, is this possible?
And if we take into account the existing manifests the `min(known_versions)` would always be `1.0.0`, please correct if i am wrong. 
Do we really need this processing logic?

I guess it will be simpler just to assume that there will always be existing manifests (that provide the known_versions).

Reply:
1. Your assumption correct.
2. The previous version of the code indeed trying to also support situation where there is not any manifest exist, and everything is coming from scratch, so I keep that behavior unchanged, and you can also see in the old test case where it assumes nothing exist from the start. The `Version(version) < Version(min(known_versions))` instead of `Version(version) < Version("1.0.0")` is to make the code as robust as possible, as there are old test cases using 0.9.0 as example.